ghost/core/core/server/data/migrations/versions/6.0/2025-06-24-09-19-42-use-object-id-for-owner-user.js

```diff
@@ -0,0 +1,196 @@
+const logging = require('@tryghost/logging');
+const ObjectID = require('bson-objectid').default;
+
+const {createTransactionalMigration} = require('../../utils');
+
+const LEGACY_OWNER_USER_ID = '1';
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        const newId = (new ObjectID()).toHexString();
+
+        logging.info(`Updating owner user ID from ${LEGACY_OWNER_USER_ID} to ${newId}`);
+
+        const currentOwnerUser = await knex('users').where('id', LEGACY_OWNER_USER_ID).first();
+
+        if (!currentOwnerUser) {
+            logging.warn(`Owner user with ID ${LEGACY_OWNER_USER_ID} not found, skipping migration`);
+
+            return;
+        }
+
+        // Verify this user actually has the Owner role
+        const ownerRole = await knex('roles').where('name', 'Owner').first();
+
+        if (!ownerRole) {
+            logging.warn('Owner role not found, skipping migration');
+
+            return;
+        }
+
+        const isOwner = await knex('roles_users')
+            .where('user_id', LEGACY_OWNER_USER_ID)
+            .where('role_id', ownerRole.id)
+            .first();
+
+        if (!isOwner) {
+            logging.warn(`User with ID ${LEGACY_OWNER_USER_ID} is not the owner, skipping migration`);
```

Comment:
If the ownership has been transferred, there may still be a user with ID 1 in the system, which I think we would still want to clean up?